VMEncryption/main/handle.py

```diff
@@ -473,38 +461,41 @@ def enable():
         hutil.do_parse_context('Enable')
         logger.log('Enabling extension')
 
+        public_settings = get_public_settings()
+        logger.log('Public settings:\n{0}'.format(json.dumps(public_settings, sort_keys=True, indent=4)))
+        cutil = CheckUtil(logger)
+
+        # run fatal prechecks, report error if exceptions are caught
+        try:
+            cutil.precheck_for_fatal_failures(public_settings)
+        except Exception as e:
+            logger.log("PRECHECK: Fatal Exception thrown during precheck")
+            logger.log(traceback.format_exc())
+            msg = e.message
+            hutil.do_exit(exit_code=0,
+                          operation='Enable',
+                          status=CommonVariables.extension_error_status,
+                          code=(CommonVariables.configuration_error),
+                          message=msg)
+
+        hutil.disk_util.log_lsblk_output()
+
         # run prechecks and log any failures detected
         try:
-            cutil = CheckUtil(logger)
-            if cutil.is_precheck_failure():
+            if cutil.is_non_fatal_precheck_failure():
                 logger.log("PRECHECK: Precheck failure, incompatible environment suspected")
             else:
                 logger.log("PRECHECK: Prechecks successful")
         except Exception:
             logger.log("PRECHECK: Exception thrown during precheck")
             logger.log(traceback.format_exc())
 
-        protected_settings_str = hutil._context._config['runtimeSettings'][0]['handlerSettings'].get('protectedSettings')
-        public_settings_str = hutil._context._config['runtimeSettings'][0]['handlerSettings'].get('publicSettings')
-
-        if isinstance(public_settings_str, basestring):
-            public_settings = json.loads(public_settings_str)
-        else:
-            public_settings = public_settings_str
-
-        if isinstance(protected_settings_str, basestring):
-            protected_settings = json.loads(protected_settings_str)
-        else:
-            protected_settings = protected_settings_str
-
-        logger.log('Public settings:\n{0}'.format(json.dumps(public_settings, sort_keys=True, indent=4)))
-
         encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
 
         if encryption_operation in [CommonVariables.EnableEncryption, CommonVariables.EnableEncryptionFormat, CommonVariables.EnableEncryptionFormatAll]:
             logger.log("handle.py found enable encryption operation")
 
-            extension_parameter = ExtensionParameter(hutil, logger, DistroPatcher, encryption_environment, protected_settings, public_settings)
+            extension_parameter = ExtensionParameter(hutil, logger, DistroPatcher, encryption_environment, get_protected_settings(), public_settings)
```

Comment:
get_public_settings() too?

Reply:
The variable public settings was populated a bit ago around line 475(old), 464(new)